test/test_framework.py

```diff
@@ -109,13 +145,16 @@ def restore(self, snapshot: typing.Dict[str, int]):
         handle = ops.Handle(None, "a_foo", "some_key")
         event = Foo(handle, 1)
 
-        framework1 = self.create_framework(tmpdir=self.tmpdir)
+        tmpdir = Path(tempfile.mkdtemp())
+        request.addfinalizer(lambda: shutil.rmtree(tmpdir))
```

Comment:
Can we do this inside `create_framework` itself?

Reply:
I think not, because many test cases create multiple frameworks sharing the same tmpdir. If the teardown is in the `create_framework` function, other frameworks might have issues accessing a non-existing directory. I have tested this already, so it has to be put outside. This was the same logic as `BaseTestCase.create_framework` where this function comes from, and I think it was designed in this way precisely because of this reason.